src/main/java/io/knowledgeassets/myskills/server/user/profile/UserProfileService.java

```diff
@@ -0,0 +1,240 @@
+package io.knowledgeassets.myskills.server.user.profile;
+
+import io.knowledgeassets.myskills.server.exception.NoSuchResourceException;
+import io.knowledgeassets.myskills.server.exception.UserProfileDocumentException;
+import io.knowledgeassets.myskills.server.exception.enums.Model;
+import io.knowledgeassets.myskills.server.user.User;
+import io.knowledgeassets.myskills.server.user.UserRepository;
+import io.knowledgeassets.myskills.server.userskill.UserSkill;
+import io.knowledgeassets.myskills.server.userskill.UserSkillRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.poi.EmptyFileException;
+import org.apache.poi.openxml4j.exceptions.NotOfficeXmlFileException;
+import org.apache.poi.xwpf.usermodel.IBody;
+import org.apache.poi.xwpf.usermodel.XWPFDocument;
+import org.apache.poi.xwpf.usermodel.XWPFParagraph;
+import org.apache.poi.xwpf.usermodel.XWPFRun;
+import org.apache.poi.xwpf.usermodel.XWPFTableCell;
+import org.apache.xmlbeans.XmlCursor;
+import org.openxmlformats.schemas.wordprocessingml.x2006.main.CTPPr;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.StreamSupport;
+
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+import static java.util.Objects.requireNonNull;
+
+@Service
+@Slf4j
+public class UserProfileService {
+
+	private static final SimpleDateFormat SDF = new SimpleDateFormat("dd.MM.yyyy");
+
+	private static final ClassPathResource DEFAULT_TEMPLATE_RESOURCE = new ClassPathResource("templates/user-profile.docx");
+
+	private final UserRepository userRepository;
+	private final UserSkillRepository userSkillRepository;
+	private final String templatePath;
+
+	public UserProfileService(UserRepository userRepository,
+							  UserSkillRepository userSkillRepository,
+							  @Value("${templates.user.profile.path:#{null}}") String templatePath) {
```

Comment:
Generally, I like to apply domain-driven principles to properties naming as well. Therefore, I typically structure the name like this:

    <my-app-name>.<feature>.<some-settings-key>

In our case, it could be something like this:

    myskills.user-profile-download.template-path

It's a matter of taste after all, I know. But following this DDD naming pattern will also allow us to easily create `@ConfigurationProperties` classes for specific features (like the document download) which can be injected to services.

Reply:
Thanks for the note! Done.